internal/handlers/registration_handler.go

```diff
@@ -0,0 +1,71 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/redhatinsights/mbop/internal/store"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+type registationCreateRequest struct {
+	UID *string `json:"uid,omitempty"`
+}
+
+func RegistrationHandler(w http.ResponseWriter, r *http.Request) {
+	b, err := io.ReadAll(r.Body)
+	if err != nil {
+		do500(w, "failed to read body bytes: "+err.Error())
+		return
+	}
+
+	var body registationCreateRequest
+	err = json.Unmarshal(b, &body)
+	if err != nil {
+		do400(w, "failed to unmarshal body: "+err.Error())
+		return
+	}
+
+	if body.UID == nil || *body.UID == "" {
+		do400(w, "required parameter [uid] not found in body")
+		return
+	}
+
+	id := identity.Get(r.Context())
+	if !id.Identity.User.OrgAdmin {
+		doError(w, "user must be org admin to register satellite", 401)
```

Comment:
Nit, but this should probably be a 403 vs a 401 since the user is authenticated, but not authorized? 
```suggestion
		doError(w, "user must be org admin to register satellite", 403)
```

Reply:
Doh. Updated!